### ISSUE TITLE
Build fails on windows as of 0.1.8 release.

### ISSUE THREAD

**Issue by b1v1r** [libp2p/go-reuseport#27]
Building with `CGO_ENABLED=0 GOOS=windows` I get the following:

```
vendor/github.com/libp2p/go-reuseport/interface.go:49: undefined: unix.ENOPROTOOPT
vendor/github.com/libp2p/go-reuseport/interface.go:60: undefined: unix.ENOPROTOOPT
vendor/github.com/libp2p/go-reuseport/interface.go:71: undefined: unix.ENOPROTOOPT
vendor/github.com/libp2p/go-reuseport/interface.go:101: undefined: unix.ENOPROTOOPT
```

Looks like some unix stuff may be sneaking into the windows build.

**Comment by Kubuxu** [comment]
Thanks for spotting this. This could have been avoided if we had fully working Windows CI, we are working on it.

@Stebalien I think correct way to resolve it is to use errors from `net` package if we can construct them.

**Comment by Stebalien** [comment]
Ah. Yes, thanks! You saved me a *lot* of trouble.

**Comment by Stebalien** [comment]
OK, this *really* sucks. If we want to use the `sys/unix` functions, we have to use the `sys/unix` types. However, `sys/windows` defines it's own copies of these types. If this were *just* go-reuseport, this would be fine (it doesn't export any of these types IIRC. However, we go-sockaddr needs to export a cross-platform interface and *does* need to export types usable by functions exported by unix and syscall.

I believe we can use the new type aliases to make this work cross-platform but that will bump our go minimum version to 1.9. Alternatively, we can export different top-level *functions* based on the platform but that's kind of a pain.

@whyrusleeping how do you feel about moving to go 1.9 over this?

**Comment by whyrusleeping** [comment]
I'm never against requiring the latest version of go.

**Comment by Stebalien** [comment]
Note: these issues are all in go-sockaddr, not go-reuseport (but it's the same problem).

**Comment by Stebalien** [comment]
Fix incoming in #28.
